### ISSUE TITLE
Class not calling through to superclass constructor when superclass has a no-arg constructor

### ISSUE THREAD

**Issue by mkpowers** [square/kotlinpoet#120]
`val typeBuilder = TypeSpec.classBuilder("Foo").superclass(Bar::class)`

This generates `class Foo : Bar`, which does not compile in Kotlin 1.1.2. I read through the documentation and the TypeSpecTests to try to get some insight, but I found nothing that indicates that I'm omitting a function call that adds the superclass constructor call.

**Comment by JakeWharton** [comment]
Related to #115.

**Comment by JakeWharton** [comment]
This should be an easy fix if you want to muster up a PR!

**Comment by hzsweers** [comment]
I can take a look at a PR. One thing that comes to mind - should you get the super classes's required constructor args automatically when you do this? Or make that an option?

**Comment by JakeWharton** [comment]
No. You have to specify them yourself which is what the other issue covers.
This would purely be for emitting parenthesis after the supertype.

**Comment by oserban** [comment]
This is still an issue in 1.0.0-RC1, as the generator does not call the zero param constructor in case of super classes. 

I can't see the code in TypeSpec anymore, so I assume the generator was refactored and this use-case was skipped.

**Comment by Egorand** [comment]
@oserban could you please share the test case that fails for you? This works for me:

```kotlin
@Test fun superclassConstructorCall() {
  val typeSpec = TypeSpec.classBuilder("MyIoException")
      .superclass(IOException::class)
      .build()
  assertThat(typeSpec.toString()).isEqualTo("""
    |class MyIoException : java.io.IOException()
    |""".trimMargin())
}
```

**Comment by oserban** [comment]
@Egorand I knew I should have provided some code samples as well. 

In all the code samples I've found online, everybody was using the _.addSuperinterface()_  method for both super classes and interfaces. I did not know about the superclass vs super interface difference. Maybe this is something worth mentioning in the current readme.

Thank you for the help.

**Comment by Egorand** [comment]
I believe this is self-explanatory, interfaces can't have constructors, so you can't expect a constructor call for a super interface.
